### PR TITLE
Introduction to RepoDb Part1

### PR DESCRIPTION
**Pull request by kchinburarat**
Document about Repodb , a hybrid-ORM library for .NET

### CODE REVIEW

**Comment by teerasej** [comment]
I fixed a little typo on this PR. But we need you for technical content @aaronamm 👍

**Comment by aaronamm** [comment]
@teerasej Thank you for your supporting. However, you can add a review comment to where there are typos and let PR owner fixt it. I think it is the prefer way.

**Comment by aaronamm** [comment]
@kchinburarat  Could you please change your branch to your origin?
image

For making a PR, we should make a new branch from our origin.
Thanks.

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+![output](images/repodb.png "result")
+
+
+อย่างที่เราทราบกันดีว่าการเข้าถึงฐานข้อมูลมีส่วนสำคัญในการพัฒนาผลิตภัณฑ์และซอฟต์แวร์ และเป็นส่วนที่สำคัญที่สุดอย่างหนึ่งของการใช้งาน เนื่องจากจำเป็นต้องได้รับการออกแบบมาเป็นอย่างดีเพื่อให้สามารถเปลี่ยนแปลงได้ง่าย
+
+ด้วยเหตุนี้ การใช้ object relational mapping (ORM) จึงเข้ามามีบทบาท เป็นพื้นฐานทั่วไปสำหรับนักพัฒนาส่วนใหญ่เมื่อพูดถึงการเข้าถึงข้อมูล ไลบรารี ORM ช่วยลดความซับซ้อนและทำให้การนำข้อมูลใน Database ไปใช้งานง่ายขึ้น และในขณะเดียวกันก็ช่วยให้การพัฒนาเร็วขึ้น มันทำงานในส่วนที่ซับซ้อนไปพร้อมกันในขณะที่เราใช้มัน แต่แน่นอนว่าเราควรทราบ "ข้อดี" และ "ข้อเสีย" ของการใช้ ORM ในแอปพลิเคชันของเราด้วย
+
+ปัจจุบันโลกของ C# (หรือ .NET โดยรวม) มีไลบรารี ORM จำนวนมากในตลาด ทั้งแบบโอเพ่นซอร์สและเชิงพาณิชย์ เฟรมเวิร์กที่ได้รับความนิยมมากที่สุดคือ Entity Framework (เฟรมเวิร์กจาก Microsoft) นอกจากนี้ยังมีส่วนอื่นๆ เช่น NHibernate, **Dapper** (micro-ORM จาก StackOverflow) และ LLBLGen Pro
+
+ด้วยเหตุนี้เอง เราในฐานะนักพัฒนาจึงมีหน้าที่เลือกสิ่งที่ดีที่สุดสำหรับความต้องการของเราเพราะเราต้องคำนึงถึง เวลาในการพัฒนา,ความเรียบง่าย,ประสิทธิภาพ ฯลฯ
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: Vey minor, not sure we need comma for Thai word. If you want to keep it, please put one space after each comma.

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ด้วยเหตุนี้ การใช้ object relational mapping (ORM) จึงเข้ามามีบทบาท เป็นพื้นฐานทั่วไปสำหรับนักพัฒนาส่วนใหญ่เมื่อพูดถึงการเข้าถึงข้อมูล ไลบรารี ORM ช่วยลดความซับซ้อนและทำให้การนำข้อมูลใน Database ไปใช้งานง่ายขึ้น และในขณะเดียวกันก็ช่วยให้การพัฒนาเร็วขึ้น มันทำงานในส่วนที่ซับซ้อนไปพร้อมกันในขณะที่เราใช้มัน แต่แน่นอนว่าเราควรทราบ "ข้อดี" และ "ข้อเสีย" ของการใช้ ORM ในแอปพลิเคชันของเราด้วย
+
+ปัจจุบันโลกของ C# (หรือ .NET โดยรวม) มีไลบรารี ORM จำนวนมากในตลาด ทั้งแบบโอเพ่นซอร์สและเชิงพาณิชย์ เฟรมเวิร์กที่ได้รับความนิยมมากที่สุดคือ Entity Framework (เฟรมเวิร์กจาก Microsoft) นอกจากนี้ยังมีส่วนอื่นๆ เช่น NHibernate, **Dapper** (micro-ORM จาก StackOverflow) และ LLBLGen Pro
+
+ด้วยเหตุนี้เอง เราในฐานะนักพัฒนาจึงมีหน้าที่เลือกสิ่งที่ดีที่สุดสำหรับความต้องการของเราเพราะเราต้องคำนึงถึง เวลาในการพัฒนา,ความเรียบง่าย,ประสิทธิภาพ ฯลฯ 
+
+##  ทำไมต้อง RepoDb ??
+![output](images/why-repodb.png "result")
+เป็นไลบรารี ORM แบบไฮบริดแบบไดนามิก ที่เบา(lightweight),มีสมรรถภาพ และมีประสิทธิภาพ(high-performance) สูงสำหรับ
+.NET
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: Should we move .NET to the same line as the above line?

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+1. คำว่า "สมรรถภาพ" หมายถึง "ความเร็ว" ORM นี้จะแปลงข้อมูลดิบเป็นอ็อบเจ็กต์คลาส และขนส่งอ็อบเจ็กต์คลาสเป็นข้อมูลจริงในฐานข้อมูล
+2. คำว่า "มีประสิทธิภาพ" หมายถึง "การจัดการที่ดี" ORM นี้ใช้หน่วยความจำคอมพิวเตอร์เมื่อจัดการกับวัตถุตลอดการใช้งาน (cycle of the process.)
+
+เราจะเรียก RepoDb ว่า hybrid-ORM ก็พราะว่ามันมีคสามสามารถที่ micro-ORM และ full-ORM มีในการที่มันทำให้เกิดความยืดหยุ่นแก่นักพัฒนาในการทำให้งานเกี่ยวกับ Database
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: Typo `คสาม` => `ความ`

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+}
+
+// Delete
+using (var connection = new SqlConnection(connectionString))
+{
+     var affectedRows = connection.ExecuteNonQuery("DELETE FROM [Person] WHERE Id = @Id;", new { Id = 10045 });
+}
+```
+
+หรือจะให้แบบการเรียก `method-based`
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: Should we change it to `หรือจะเรียกใช้แบบ`?

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+```cs
+// Query
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md:75]: Very minor, I am not sure we should change `Query` to `Select`.

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+using (var connection = new SqlConnection(connectionString))
+{
+     var deletedRows = connection.Delete<Person>(p => p.Id == 10045);
+}
+```
+
+ฟีเจอร์อื่นๆ ของ ORM แบบอื่นยังมีอยู่ในไลบรารีนี้ผ่านการเรียกใช้ตามวิธีการ (เช่น BatchQuery, BulkInsert, Delete, DeleteAll, Insert, InsertAll, Update, UpdateAll, Merge, MergeAll เป็นต้น)
+
+## Batch Operations
+การดำเนินการแบบแบตช์ในไลบรารีนี้เป็นคำสั่งแบบแพ็กทั้งหมด โดย DbCommand เดียวใช้เพื่อดำเนินการคำสั่ง SQL หลายรายการในครั้งเดียว กระบวนการนี้เป็น ACID โดยค่า default chunk คือ 10 records ลองดูตัวอย่างด้านล่าง
```
Comment (aaronamm) [inline content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: Could you please add parentheses of what ACID stand for and change it to a useful link that explains what ACID is?

**Review by aaronamm — CHANGES_REQUESTED** [review]
@kchinburarat very nice PR, keep going.
I have only minor feedback. After you have fixed them, I will merge your PR.
BTW, is it possible for you to put some content that why we need to consider Repodb what is its benefit over Dapper.
I haven't use it but the raw query API looks similar to Dapper.

Thanks.

**Comment by teerasej** [comment]
> @teerasej Thank you for your supporting. However, you can add a review comment to where there are typos and let PR owner fixt it. I think it is the prefer way.

ok, sorry about that 👍

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 5 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md:75]: We do select but the method name is Query so I prefer to use the word `Query` :)

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 6 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: fixed

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 4 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: fixed

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 3 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: fixed

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 2 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: fixed

File: content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md (hunk 1 above)
Comment (kchinburarat) [reply content/faq/libraries-frameworks/orm/RepoDb/repodb-hybrid-orm-part1.md]: fixed

**Comment by aaronamm** [comment]
@kchinburarat Good job. Thanks.

**Review by aaronamm — APPROVED** [review]

**Pull request merged**